### PR TITLE
Lint: ineffective access modifier

### PR DESCRIPTION
**Pull request by gooallen**
Related to #84
 - protected access modifiers don't make singleton methods

### CODE REVIEW

**Comment by cttttt** [comment]
@gooallen based on [the documentation](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Lint/IneffectiveAccessModifier) the linter error is warning that putting this class method under an access modifier just won't do anything.  They suggest using `:private_class_method`.

**Comment by gooallen** [comment]
@cttttt I had tried `:private_class_method` from the documentation and didn't work out.

**Comment by XaF** [comment]
What was the complaint? Can you submit a PR using it so we can see what's going on?

File: lib/record_store/record.rb [hunk 1]
```diff
@@ -86,16 +86,18 @@ def to_s
       "[#{type}Record] #{fqdn} #{ttl} IN #{type} #{rdata_txt}"
     end
 
+    def self.ensure_ends_with_dot(fqdn)
+      fqdn.end_with?(".") ? fqdn : "#{fqdn}."
+    end
+
+    private_class_method :method
```
Comment (XaF) [inline lib/record_store/record.rb]: What is `method`? 
```suggestion
    private_class_method :ensure_ends_with_dot
```

**Comment by gooallen** [comment]
This is the error log that I mentioned
```
Mocha deprecation warning at /opt/rubies/2.4.3/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:130:in `require': `require 'mocha/mini_test'` has been deprecated. Please use `require 'mocha/minitest' instead.
/Users/byeongyun/.gem/ruby/2.4.3/gems/httpclient-2.8.3/lib/httpclient/ssl_config.rb:51: warning: method redefined; discarding old initialize
/Users/byeongyun/.gem/ruby/2.4.3/gems/httpclient-2.8.3/lib/httpclient/ssl_config.rb:58: warning: method redefined; discarding old add_cert
/Users/byeongyun/.gem/ruby/2.4.3/gems/httpclient-2.8.3/lib/httpclient/ssl_config.rb:58: warning: method redefined; discarding old add_file
/Users/byeongyun/.gem/ruby/2.4.3/gems/httpclient-2.8.3/lib/httpclient/ssl_config.rb:58: warning: method redefined; discarding old add_path
<compiled>: warning: possibly useless use of == in void context
/Users/byeongyun/.gem/ruby/2.4.3/gems/google-api-client-0.23.9/lib/google/apis/core/batch.rb:169: warning: assigned but unused variable - length
/Users/byeongyun/.gem/ruby/2.4.3/gems/google-api-client-0.23.9/lib/google/apis/core/base_service.rb:215: warning: method redefined; discarding old client
/Users/byeongyun/.gem/ruby/2.4.3/gems/zonefile-1.06/lib/zonefile/zonefile.rb:219: warning: assigned but unused variable - data
/Users/byeongyun/.gem/ruby/2.4.3/gems/zonefile-1.06/lib/zonefile/zonefile.rb:227: warning: assigned but unused variable - name
/Users/byeongyun/.gem/ruby/2.4.3/gems/zonefile-1.06/lib/zonefile/zonefile.rb:227: warning: assigned but unused variable - dclass
/Users/byeongyun/.gem/ruby/2.4.3/gems/zonefile-1.06/lib/zonefile/zonefile.rb:227: warning: assigned but unused variable - type
/Users/byeongyun/.gem/ruby/2.4.3/gems/zonefile-1.06/lib/zonefile/zonefile.rb:227: warning: assigned but unused variable - host
/Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/record.rb:33:in `initialize': private method `ensure_ends_with_dot' called for RecordStore::Record:Class (NoMethodError)
	from /Users/byeongyun/src/github.com/Shopify/record_store/lib/record_store/record/alias.rb:9:in `initialize'
	from /Users/byeongyun/src/github.com/Shopify/record_store/test/record_test.rb:5:in `new'
	from /Users/byeongyun/src/github.com/Shopify/record_store/test/record_test.rb:5:in `<class:RecordTest>'
	from /Users/byeongyun/src/github.com/Shopify/record_store/test/record_test.rb:3:in `<top (required)>'
	from /opt/rubies/2.4.3/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /opt/rubies/2.4.3/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/byeongyun/.gem/ruby/2.4.3/gems/rake-12.3.3/lib/rake/rake_test_loader.rb:17:in `block in <main>'
	from /Users/byeongyun/.gem/ruby/2.4.3/gems/rake-12.3.3/lib/rake/rake_test_loader.rb:5:in `select'
	from /Users/byeongyun/.gem/ruby/2.4.3/gems/rake-12.3.3/lib/rake/rake_test_loader.rb:5:in `<main>'
rake aborted!
Command failed with status (1)
/Users/byeongyun/.gem/ruby/2.4.3/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

**Pull request merged**